src/parse_instruction.py

```diff
@@ -237,6 +237,24 @@ class NaiveParsingArch(ArchAsmParsing):
     def normalize_instruction(self, instr: AsmInstruction) -> AsmInstruction:
         return instr
 
+@dataclass
+class UsedRegNames:
+    used_names: Dict[Register, Register] = field(default_factory=dict)
+
+    def input_to_internal(self, reg: Register, arch: ArchAsmParsing) -> Register:
+        if reg in arch.aliased_regs:    
+            if reg not in self.used_names.values():
+                self.used_names[arch.aliased_regs[reg]] = reg
```

Comment:
```suggestion
            self.used_names[arch.aliased_regs[reg]] = reg
```
(equivalent but O(1) instead of O(n))